bloom-housing/bloom
# fix: validate url field in building section criteria

# Pull Request Template

## Issue Overview

This PR addresses #3422 


## Description

its part of task that will trigger validation from `url` type field. 
Needs UIC change from [here](https://github.com/bloom-housing/ui-components/pull/113)
Additionally `Open house` link field had that bug so i fixed this one also.
Added validation for `/settings` section on partners, both for preference url, and option url.
`Custom Online Application URL` is not validated, as there is an issue with it reseting on submit form, that field can be validate later on backend side ([ticket](https://github.com/bloom-housing/bloom/issues/3510))

## How Can This Be Tested/Reviewed?

Go to partners site edit listing and `building section criteria` url will now be validated on save.
Also pdf save part of `building section criteria` should still work like before.
Additionally validation for `Open house` link will now require `https` or `http`, and will also have updated error message.
In `/settings` when creating item check url fields for preference, and option (add option)

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
